Fix bug 1429431: Use pyup to monitor and update e2e-tests dependencies.

Osmose opened the pull request:
This PR does a few things:

1. Moves the dependency config for the e2e-tests into `e2e-tests/requirements/default.txt` and adds a constriants file for indirect dependencies. Both of these files perform hash checks for downloaded packages.
2. Adds the newly-created e2e-tests requirements to the pyup config.
3. Removes the `flake8` environment from the tox config, instead using the existing automation we have for linting python code in CircleCI and TravisCI. I'm not sure if this is being actively run anywhere.

   The downside of this is that we can't ignore line length only for the e2e-tests subdirectory, so I fixed the linting errors that this change brought up.

@stephendonner @chartjes Ya'll probably wanna look over this too before we merge it in case I've unknowingly broken some automation I'm not aware of or am otherwise messing up your stuff. This change will make it so that a PR for updating the test dependencies will be automatically filed once a month by the pyup bot, saving ya'll some human effort. :D

willkg commented on e2e-tests/pages/super_search_page.py:
```diff
@@ -101,13 +119,22 @@ def select_match(self, line_id, match):
+        return self.find_element(
+            self._facet_text_locator[0],
+            self._facet_text_locator[1] % line_id,
+        ).text
```
I probably would have done this instead:

```
elem = self.find_element(
    self._facet_text_locator[0],
    self._facet_text_locator[1] % line_id
)
return elem.text
```

Seems better to break it up with an intermediary step. It's fine as is, too.

willkg commented:
@stephendonner @chartjes Can you two weigh in here regarding the PR description questions?

chartjes commented:
@willkg From my end all this looks fine -- it will not currently break anything that I am aware of

stephendonner commented on e2e-tests/tox.ini:
```diff
@@ -1,37 +1,21 @@
 [tox]
 skipsdist = true
-envlist = py27, flake8
+envlist = py27
```
Removing flake8 from this Tox env here breaks our Jenkins build, due to https://github.com/mozilla-services/socorro/blob/87eb5aa4d85e355739b1db13df729e2675ceba99/e2e-tests/Jenkinsfile#L29 - that 'Lint' stage should be removed.

stephendonner replied:
r+ once the Tox/flake8/Jenkinsfile issue has been addressed

stephendonner commented:
Mentioning @davehunt to make him aware of the changes (particularly to the flake8 Tox env).

davehunt commented:
Thanks @stephendonner 

I'm okay with losing the flake8 tox environment, though it would be good to know if we're running isort in TravisCI/CircleCI so we're not losing that coverage. Adhering to the project's line length makes sense too.

I'm interested to know if there's any chance of running the e2e tests elsewhere? Does socorro have another Jenkins instance, or is everything run via Circle/Travis? Is there a way we can increase the awareness and dev ownership of these tests? I'm interested in your thoughts.

Oh, and we're moving many of our automation projects away from Tox, so you can expect a followup on that soon.

Osmose commented:
Updated with the removal of the Jenkins lint job.

> I'm okay with losing the flake8 tox environment, though it would be good to know if we're running isort in TravisCI/CircleCI so we're not losing that coverage. Adhering to the project's line length makes sense too.

I added a commit that adds this to our flake8 linting only for the e2e-tests directory. Running it on the rest of the project yielded about 1500 lint errors. So, er, we'll just ignore them for now.
 
> I'm interested to know if there's any chance of running the e2e tests elsewhere? Does socorro have another Jenkins instance, or is everything run via Circle/Travis? Is there a way we can increase the awareness and dev ownership of these tests? I'm interested in your thoughts.

I've been working on this a bit when I'm not working on my main project for H1 (Firefox JS error collection); #4290 just landed, which fixes one of the blockers for running these tests in automation instead of only against already-set-up environments.

Osmose commented:
Re-asking for review since I'm unsure if we're okay adding all those new deps just for isort on the e2e-tests.
